datalake/src/main/java/com/sequenceiq/datalake/job/SdxClusterStatusCheckerJob.java

```diff
@@ -172,14 +177,27 @@ private void handleRunningSdx(StackStatusV4Response stack, SdxCluster sdx) {
             setDeleteFailed(stack, sdx);
         } else if (stack.getStatus() == Status.DELETED_ON_PROVIDER_SIDE) {
             setDeletedOnProviderSide(stack, sdx);
-        } else if (stack.getStatus() == Status.AMBIGUOUS || (stack.getStatus() == Status.AVAILABLE && stack.getClusterStatus() == Status.AMBIGUOUS)) {
-            String statusReason = stack.getStatus() == Status.AMBIGUOUS ? stack.getStatusReason() : stack.getClusterStatusReason();
-            sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.CLUSTER_AMBIGUOUS, ResourceEvent.CLUSTER_AMBARI_CLUSTER_SYNCHRONIZED,
+        } else if (isUnreachable(stack.getStatus(), stack.getClusterStatus())) {
+            String statusReason = stack.getStatus() == Status.UNREACHABLE ? stack.getStatusReason() : stack.getClusterStatusReason();
+            sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.CLUSTER_UNREACHABLE, ResourceEvent.CLUSTER_AMBARI_CLUSTER_SYNCHRONIZED,
                     Collections.singleton(sdx.getClusterName()), statusReason, sdx);
-            logStateChange(DatalakeStatusEnum.RUNNING, DatalakeStatusEnum.CLUSTER_AMBIGUOUS);
+            logStateChange(DatalakeStatusEnum.RUNNING, DatalakeStatusEnum.CLUSTER_UNREACHABLE);
+        } else if (isNodeFailure(stack.getStatus(), stack.getClusterStatus())) {
+            String statusReason = stack.getStatus() == Status.NODE_FAILURE ? stack.getStatusReason() : stack.getClusterStatusReason();
+            sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.NODE_FAILURE, ResourceEvent.CLUSTER_AMBARI_CLUSTER_SYNCHRONIZED,
+                    Collections.singleton(sdx.getClusterName()), statusReason, sdx);
+            logStateChange(DatalakeStatusEnum.RUNNING, DatalakeStatusEnum.NODE_FAILURE);
         }
     }
 
+    private boolean isUnreachable(Status stackStatus, Status clusterStatus) {
```

Comment:
So, here:
- stack could be: NODE_FAILURE, UNREACHABLE
- cluster could be: NODE_FAILURE, UNREACHABLE

In case of stack what is the difference between NODE_FAILURE and UNREACHABLE? Shouldn't it only be REACHABLE or UNREACHABLE? Or are we partitioning stackStatus like:

 UNREACHABLE: problem thorough provider API
 NODE_FAILURE: CCM or nginx does no work?

Reply:
I see that this comes from the older implementation as probably both statuses were set in conjunction.

Reply:
UNREACHABLE: -> We cannot reach, maybe CM, NGNIX, CP failed
NODE_FAILURE: -> CM, NGNIX, CP works, but it is still not working properly e.g. CM agent failed, Entire VM failed, or even just a service failed.